src/gpu/intel/sycl/l0/utils.cpp

```diff
@@ -272,6 +287,113 @@ bool compare_ze_devices(const ::sycl::device &lhs, const ::sycl::device &rhs) {
     return lhs_ze_handle == rhs_ze_handle;
 }
 
+status_t get_device_ip(ze_device_handle_t device, uint32_t &ip_version) {
+    ze_device_properties_t deviceProps = {ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES};
+    ze_device_ip_version_ext_t devicePropsIP
+            = {ZE_STRUCTURE_TYPE_DEVICE_IP_VERSION_EXT};
+    deviceProps.pNext = &devicePropsIP;
+    CHECK(func_zeDeviceGetProperties(device, &deviceProps));
+    ip_version = devicePropsIP.ipVersion;
+    return status::success;
+}
+
+status_t get_l0_device_enabled_systolic_intel(
+        ze_device_handle_t device, bool &mayiuse_systolic) {
+    ze_device_module_properties_t deviceModProps
+            = {ZE_STRUCTURE_TYPE_DEVICE_MODULE_PROPERTIES};
+    // Note: supported by Intel Driver 24.05 and onwards
+    ze_intel_device_module_dp_exp_properties_t deviceModPropsExt
+            = {ZE_STRUCTURE_INTEL_DEVICE_MODULE_DP_EXP_PROPERTIES};
+    deviceModProps.pNext = &deviceModPropsExt;
+
+    CHECK(func_zeDeviceGetModuleProperties(device, &deviceModProps));
+    mayiuse_systolic
+            = deviceModPropsExt.flags & ZE_INTEL_DEVICE_MODULE_EXP_FLAG_DPAS;
+    return status::success;
+}
+
+status_t get_l0_device_enabled_native_float_atomics(
+        ze_device_handle_t device, uint64_t native_extensions) {
+    using namespace gpu::intel::compute;
+
+    ze_device_properties_t deviceProps = {ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES};
+    ze_float_atomic_ext_properties_t fltAtom
+            = {ZE_STRUCTURE_TYPE_FLOAT_ATOMIC_EXT_PROPERTIES};
+    deviceProps.pNext = &fltAtom;
+    CHECK(func_zeDeviceGetProperties(device, &deviceProps));
+
+    ze_device_fp_atomic_ext_flags_t atomic_load_store
+            = ZE_DEVICE_FP_ATOMIC_EXT_FLAG_GLOBAL_LOAD_STORE
+            | ZE_DEVICE_FP_ATOMIC_EXT_FLAG_LOCAL_LOAD_STORE;
+    ze_device_fp_atomic_ext_flags_t atomic_add
+            = ZE_DEVICE_FP_ATOMIC_EXT_FLAG_GLOBAL_ADD
+            | ZE_DEVICE_FP_ATOMIC_EXT_FLAG_LOCAL_ADD;
```

Comment:
No GPUs currently have floating point atomic adds in local memory, so we just need to know about global memory FP atomic support:
```suggestion
            = ZE_DEVICE_FP_ATOMIC_EXT_FLAG_GLOBAL_ADD;
```
I think a similar change makes sense for load/store and min/max.

Reply:
I aligned with the `native_extension` definition that was used so far from OCL queries [here](https://github.com/oneapi-src/oneDNN/blob/316287183e2503a9db2d975dd04359728aabceb2/src/gpu/intel/ocl/ocl_utils.cpp#L394) (flag is set only if both local and global are available).

I am fine changing the semantic of `native_extensions` (it seems broken anyway as you mention). But that might have wider implications.

Reply:
Good point. Then I think it makes sense to keep L0 and OCL aligned and change both in a future PR.